src/checks/kubernetes/services/ports/ports.go

```diff
@@ -0,0 +1,80 @@
+package ports
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	"gopkg.in/yaml.v2"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+var kubeClientSet *kubernetes.Clientset
+
+type inputs struct {
+	valuesYaml string
+	checkName  string
+	namespace  string
+}
+
+type Results struct {
+	DidPass bool
+	Message string
+}
+
+// New New
+func New(valuesYaml string, clientSet *kubernetes.Clientset, checkName string, namespace string) inputs {
+	s := inputs{valuesYaml, checkName, namespace}
+	kubeClientSet = clientSet
+
+	return s
+}
+
+type doesPortExistStruct struct {
+	Values struct {
+		ServiceName string `yaml:"serviceName"`
+		Port        int32  `yaml:"port"`
+	} `yaml:"values"`
+}
+
+// DoesPortExist DoesPortExist
+func (i inputs) DoesPortExist() Results {
+
+	var values doesPortExistStruct
+
+	err := yaml.Unmarshal([]byte(i.valuesYaml), &values)
+	if err != nil {
+		log.Fatalf("Unmarshal: %v", err)
+	}
+
+	// Set initial check results
+	checkResult := Results{
+		DidPass: false,
+		Message: "Port not found: " + fmt.Sprint(values.Values.Port),
+	}
+
+	// Run kube stuff
+	services, err := kubeClientSet.CoreV1().Services(i.namespace).List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		panic(err.Error())
+	}
+
+	for _, aService := range services.Items {
+
+		if aService.ObjectMeta.Name == values.Values.ServiceName {
+			fmt.Println("Found service: " + aService.ObjectMeta.Name)
+
+			for _, port := range aService.Spec.Ports {
+				if port.Port == values.Values.Port {
+					fmt.Println("Found port: " + fmt.Sprint(values.Values.Port))
+
+					checkResult.DidPass = true
+					checkResult.Message = "Port found: " + fmt.Sprint(values.Values.Port)
+				}
+			}
+		}
+	}
```

Comment:
I like this, we should extend it to also check for a clusterIP and endpoints to verify that there are backends the service can reach.

Reply:
@mazamats yes, i think that would be useful.  Should this be one check or split up into different checks?

Reply:
One check would be easier imo.

What do you think of  renaming `servicePort` to just `service` and adding params for testing various aspects of the service?

Example:

```yaml
- ttype: serviceCheck
   name: foobar-service-check
   namespace: hos-m2
   # What to test
   tests:
     ports: true
     endpoints: true
     clusterIP: true
     hostPort: false
   # Input values for this specific check
   values:
     serviceName: healthapp-caregaps
     port: 20014
```

Reply:
Gotcha...i like it!  

I am going to capture this in another ticket to add this.  Since there are no major objections with the structure of this PR or fundamental to changing something, I would like to merge this first so that subsequent adds/changes can be an easier PR.

Reply:
sounds good

Reply:
Captured this information here: https://anthemai.atlassian.net/browse/HSS-209